src/time_integration/methods_SSP.jl

```diff
@@ -0,0 +1,236 @@
+  function SimpleSSPRK33(; stage_callbacks=(AntidiffusiveStage(),))
+    a = SVector(0.0, 3/4, 1/3)
+    b = SVector(1.0, 1/4, 2/3)
+    c = SVector(0.0, 1.0, 1/2)
+
+    # Butcher tableau
+    #   c |       a
+    #   0 |
+    #   1 |   1
+    # 1/2 | 1/4  1/4
+    # --------------------
+    #   b | 1/6  1/6  2/3
+
+    new{typeof(stage_callbacks)}(a, b, c, stage_callbacks)
+  end
+end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L1
+mutable struct SimpleIntegratorSSPOptions{Callback}
+  callback::Callback # callbacks; used in Trixi
+  adaptive::Bool # whether the algorithm is adaptive; ignored
+  dtmax::Float64 # ignored
+  maxiters::Int # maximal number of time steps
+  tstops::Vector{Float64} # tstops from https://diffeq.sciml.ai/v6.8/basics/common_solver_opts/#Output-Control-1; ignored
+end
+
+function SimpleIntegratorSSPOptions(callback, tspan; maxiters=typemax(Int), kwargs...)
+  SimpleIntegratorSSPOptions{typeof(callback)}(
+    callback, false, Inf, maxiters, [last(tspan)])
+end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L77
+# This implements the interface components described at
+# https://diffeq.sciml.ai/v6.8/basics/integrator/#Handing-Integrators-1
+# which are used in Trixi.
+mutable struct SimpleIntegratorSSP{RealT<:Real, uType, Params, Sol, F, Alg, SimpleIntegratorSSPOptions}
+  u::uType
+  du::uType
+  r0::uType
+  t::RealT
+  dt::RealT # current time step
+  dtcache::RealT # ignored
+  iter::Int # current number of time steps (iteration)
+  p::Params # will be the semidiscretization from Trixi
+  sol::Sol # faked
+  f::F
+  alg::Alg
+  opts::SimpleIntegratorSSPOptions
+  finalstep::Bool # added for convenience
+end
+
+# Forward integrator.stats.naccept to integrator.iter (see GitHub PR#771)
+function Base.getproperty(integrator::SimpleIntegratorSSP, field::Symbol)
+  if field === :stats
+    return (naccept = getfield(integrator, :iter),)
+  end
+  # general fallback
+  return getfield(integrator, field)
+end
+
+"""
+    solve(ode; dt, callbacks, kwargs...)
+
+The following structures and methods provide a implementation of the third-order SSP Runge-Kutta
+method [`SimpleSSPRK33`](@ref).
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+function solve(ode::ODEProblem; alg=SimpleSSPRK33()::SimpleAlgorithmSSP,
+               dt, callback=nothing, kwargs...)
+  u = copy(ode.u0)
+  du = similar(u)
+  r0 = similar(u)
+  t = first(ode.tspan)
+  iter = 0
+  integrator = SimpleIntegratorSSP(u, du, r0, t, dt, zero(dt), iter, ode.p,
+                  (prob=ode,), ode.f, alg,
+                  SimpleIntegratorSSPOptions(callback, ode.tspan; kwargs...), false)
+
+  # resize container
+  resize!(integrator.p, nelements(integrator.p.solver, integrator.p.cache))
```

Comment:
This is the only place that poses a problem for using OrdinaryDiffEq's time integration methods (at least for the MCL coming in later PRs). I.e., resizing the containers at the start of the simulation. I found in Trixi the function `reinitialize_containers!` which seems to do this for the other containers. Would it make sense to include this call there?

Reply:
Why does this need to be resized here? And is this the only reason why the entire the integrator is implemented here?

Reply:
The containers are created with size 0 in this entry, since at this point of initialization no mesh or semidiscretization exists. So, I have to resize it before the simulation.

Reply:
There are other issues. The IDP limiting requires for example the antidiffusive stage, which has to take place between the low-order update and the convex combination step.
The situation is different for MCL. It is a-priori and does not need this antidiffusive stage. So, it would be possible to simulate it also with OrdinaryDiffEq, if one accommodates this Resize call elsewhere.

Reply:
Usually, all callbacks have a ˋinitialize!ˋ function that is called by ˋsolveˋ at the beginning before the time loop starts.Couldn’t you do the resizing there?

Reply:
Unfortunately, there is no “normal” step callback that must necessarily be executed. So, I don't think that is useful. 

If I added it to one of my “new” stage callbacks, e.g., the antidiffusive stage, we would have gained nothing, as they cannot be used by OrdinaryDiffEq.

Reply:
> This is the only place that poses a problem for using OrdinaryDiffEq

Then I misunderstood your statement above. I thought that this initialization is the only reason why you need to implement your own time integrator, but there are other reasons, as it seems?

Reply:
I tried to explain it above.
For the MCL approach, it is theoretically possible to use OrdinaryDiffEq's time methods, since no antidiffusive stage is used. But this limiter is not yet in this PR. Therefore, it probably makes sense to address the issue later, when it becomes important.

The IDP limiting, as used in this PR, necessarily requires the antidiffusive stage and therefore external time methods cannot be used.